### DISCUSSION TITLE
Is there any way to know how many times the message was resent?

### DISCUSSION THREAD

**Discussion by spaiz** [rabbitmq/rabbitmq-server#2994] [Questions] [answered]
Let's assume I have some message, and sometimes my consumer crashes, so the message will be resent by RabbitMQ (cause the message was not ack/nack).

Is there any way to know how many times the message was resent? The main idea is to be able to build some graph to catch situations like "Pay attention, this message was resent X times".

**Comment by johanrhodin** [comment, marked as answer]
In later versions of RabbitMQ this is possible for QQ: [poison message handling](https://www.rabbitmq.com/quorum-queues.html#poison-message-handling)
It is not possible for classic queues (today).
